Run BK exhaustive pipeline when code is pushed

dliappis opened the pull request:
<!-- Type of change
Please label this PR with the release version and one of the following labels, depending on the scope of your change:
- bug
- enhancement
- breaking change
- doc
-->

## Release notes
[rn:skip]

## What does this PR do?

As we are close to sunsetting Jenkins by completing the exhaustive tests Buildkite pipeline, it makes sense to start already running the existing phases (testing and compatibility).

This commit defines the trigger to be code events, i.e. direct pushes and merge commits, similar to what we currently have in Jenkins.

## Why is it important/What is the impact to the user?

It allows us to gradually remove the corresponding phases from Jenkins to reduce CI duplication.
This will be especially useful when merging https://github.com/elastic/logstash/pull/15696 as the change is incompatible with Jenkins.

## How to test this PR locally

An example build triggered manually can be found in https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/49#_
Unfortunately Windows tests will fail due to https://github.com/elastic/logstash/issues/15562

## Related issues

- Relates https://github.com/elastic/ingest-dev/issues/1722

dliappis commented on catalog-info.yaml:
```diff
@@ -436,11 +436,15 @@ spec:
-        trigger_mode: none
-      cancel_intermediate_builds: true
-      skip_intermediate_builds: true
+        build_branches: true
```
this means that in addition to merge commits, the pipeline will be triggered when we push a new branch, e.g. `8.14`. I don't know if this is something we want, it felt right to me.

jsvd replied:
Agreed, as long as we exclude `docs` changes (if possible)

dliappis commented on catalog-info.yaml:
```diff
@@ -436,11 +436,15 @@ spec:
+        build_branches: true
+        build_pull_request_forks: false
+        build_pull_requests: false
+        build_tags: false
```
conversely, I felt that we don't need to trigger this pipeline when we push tags.

jsvd replied:
Agreed, not for tags.

dliappis commented on catalog-info.yaml:
```diff
@@ -436,11 +436,15 @@ spec:
+        build_tags: false
+        trigger_mode: code
+      cancel_intermediate_builds: false
+      skip_intermediate_builds: false
```
I'd like to discuss what we want to do here; the above two lines `cancel_intermediate_builds` / `skip_intermediate_builds`, when set to `true` cancel currently running jobs (per branch) when a new commit is pushed.

This makes sense e.g. for PRs, but for intake style pipelines like this one, I think we want to let each job run completely, even if a subsequent commit lands soon after.

We can also tune this behavior per branch; e.g. [elastic-agent](https://github.com/elastic/elastic-agent/blob/main/catalog-info.yaml#L51) sets to `true` by default, but only for non `main` branches:

```
      cancel_intermediate_builds: true
      cancel_intermediate_builds_branch_filter: "!main"
      skip_intermediate_builds: true
      skip_intermediate_builds_branch_filter: "!main"
```

jsvd replied:
it's fine that we don't cancel or skip for non-PRs, rarely we have a lot of commits landing on a given branch, so we can afford to have each commit properly/individually tested.

dliappis replied:
Ok I'll keep things as they are then, thanks.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/609)
* Commit: faf81b720725ee3562b5c9c1078e5f55fb95abbd

cc @dliappis

<!--buildkite-pr-comment-logstash-pull-request-pipeline
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/609","number":609,"commit":"faf81b720725ee3562b5c9c1078e5f55fb95abbd"}],"number":609}
buildkite-pr-comment-->

dliappis commented:
@elasticmachine run elasticsearch-ci/docs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/654)
* Commit: cd30f0f3c2ead12061bbb17122d3c56ff4c0fedb

### Failed CI Steps
* [:lab_coat: Integration Tests / part 2](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/654#018cee81-92bd-45dc-b017-44c941ecd43d)
* [:lab_coat: IT Persistent Queues / part 2](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/654#018cee81-92c5-44b0-a23b-d2b5142c5325)

### History
* :green_heart: [Build #609](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/609) succeeded faf81b720725ee3562b5c9c1078e5f55fb95abbd

cc @dliappis

<!--buildkite-pr-comment-logstash-pull-request-pipeline
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/654","number":654,"commit":"cd30f0f3c2ead12061bbb17122d3c56ff4c0fedb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/609","number":609,"commit":"faf81b720725ee3562b5c9c1078e5f55fb95abbd"}],"number":654}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/656)
* Commit: 9204c0969b1637c447703b905efb7a50d341a124

### Failed CI Steps
* [:lab_coat: Integration Tests / part 1](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/656#018cee87-a9ea-41a9-82f8-2a8ae9fa5dc9)

### History
* :broken_heart: [Build #654](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/654) failed cd30f0f3c2ead12061bbb17122d3c56ff4c0fedb
* :green_heart: [Build #609](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/609) succeeded faf81b720725ee3562b5c9c1078e5f55fb95abbd

cc @dliappis

<!--buildkite-pr-comment-logstash-pull-request-pipeline
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/656","number":656,"commit":"9204c0969b1637c447703b905efb7a50d341a124"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/654","number":654,"commit":"cd30f0f3c2ead12061bbb17122d3c56ff4c0fedb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/609","number":609,"commit":"faf81b720725ee3562b5c9c1078e5f55fb95abbd"}],"number":656}
buildkite-pr-comment-->

dliappis commented:
@jsvd I addressed your comment (about skipping on docs changes). This is an example (manually triggered) build: https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/56

Regarding the other comments we aligned and no further changes are needed. Please take another look when you can.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/658)
* Commit: 71cf40df44315dc063eaf0b029c789e14d44d37a

### Failed CI Steps
* [:lab_coat: IT Persistent Queues / part 2](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/658#018ceea3-43be-42d8-b426-e603ccce5f7c)

### History
* :broken_heart: [Build #656](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/656) failed 9204c0969b1637c447703b905efb7a50d341a124
* :broken_heart: [Build #654](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/654) failed cd30f0f3c2ead12061bbb17122d3c56ff4c0fedb
* :green_heart: [Build #609](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/609) succeeded faf81b720725ee3562b5c9c1078e5f55fb95abbd

cc @dliappis

<!--buildkite-pr-comment-logstash-pull-request-pipeline
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/658","number":658,"commit":"71cf40df44315dc063eaf0b029c789e14d44d37a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/656","number":656,"commit":"9204c0969b1637c447703b905efb7a50d341a124"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/654","number":654,"commit":"cd30f0f3c2ead12061bbb17122d3c56ff4c0fedb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/609","number":609,"commit":"faf81b720725ee3562b5c9c1078e5f55fb95abbd"}],"number":658}
buildkite-pr-comment-->

dliappis commented:
@logstashmachine backport 8.12
